File: internal/domain/group.go
package domain

import "time"

type LoadBalanceStrategy string

const (
	StrategyRoundRobin       LoadBalanceStrategy = "round_robin"
	StrategyWeighted         LoadBalanceStrategy = "weighted"
	StrategyLeastConnections LoadBalanceStrategy = "least_connections"
	StrategyPriority         LoadBalanceStrategy = "priority"
	StrategyAdaptive         LoadBalanceStrategy = "adaptive"
)

type Group struct {
	ID                  string              `json:"id" gorm:"primaryKey;size:64"`
	Name                string              `json:"name" gorm:"size:255;not null;uniqueIndex"`
	Description         string              `json:"description" gorm:"type:text"`
	LoadBalanceStrategy LoadBalanceStrategy `json:"load_balance_strategy" gorm:"size:64;not null;default:round_robin"`
	RateLimitConfig     map[string]any      `json:"rate_limit_config" gorm:"serializer:json"`
	QuotaConfig         map[string]any      `json:"quota_config" gorm:"serializer:json"`
	Enabled             bool                `json:"enabled" gorm:"default:true"`
	Version             int                 `json:"version" gorm:"not null;default:0"`
	Metadata            map[string]any      `json:"metadata" gorm:"serializer:json"`
	CreatedAt           time.Time           `json:"created_at"`
	UpdatedAt           time.Time           `json:"updated_at"`
}

func (Group) TableName() string {
	return "groups"
}

// GroupModel 分组-模型绑定关系
type GroupModel struct {
	ID        string    `json:"id" gorm:"primaryKey;size:64"`
	GroupID   string    `json:"group_id" gorm:"size:64;not null;uniqueIndex:idx_group_model"`
	ModelID   string    `json:"model_id" gorm:"size:64;not null;uniqueIndex:idx_group_model"`
	Enabled   bool      `json:"enabled" gorm:"default:true"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (GroupModel) TableName() string {
	return "group_models"
}

File: ui/src/api/group.ts
import { getAuthHeaders } from './auth'

const API_BASE = '/admin/v1'

export interface Group {
  id: string
  name: string
  description: string
  load_balance_strategy: string
  rate_limit_config: Record<string, any>
  quota_config: Record<string, any>
  enabled: boolean
  version: number
  metadata: Record<string, any>
  created_at: string
  updated_at: string
  model_ids?: string[]
}

export interface GroupModel {
  id: string
  group_id: string
  model_id: string
  enabled: boolean
  created_at: string
  updated_at: string
}

export async function fetchGroups(): Promise<{ items: Group[]; total: number }> {
  const res = await fetch(`${API_BASE}/groups`, {
    headers: getAuthHeaders(),
  })
  if (!res.ok) throw new Error('Failed to fetch groups')
  return res.json()
}

export async function fetchGroup(id: string): Promise<Group> {
  const res = await fetch(`${API_BASE}/groups/${id}`, {
    headers: getAuthHeaders(),
  })
  if (!res.ok) throw new Error('Failed to fetch group')
  return res.json()
}

export async function createGroup(data: Partial<Group>): Promise<Group> {
  const res = await fetch(`${API_BASE}/groups`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      ...getAuthHeaders(),
    },
    body: JSON.stringify(data),
  })
  if (!res.ok) throw new Error('Failed to create group')
  return res.json()
}

export async function updateGroup(id: string, data: Partial<Group>): Promise<Group> {
  const res = await fetch(`${API_BASE}/groups/${id}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
      ...getAuthHeaders(),
    },
    body: JSON.stringify(data),
  })
  if (!res.ok) throw new Error('Failed to update group')
  return res.json()
}

export async function deleteGroup(id: string): Promise<void> {
  const res = await fetch(`${API_BASE}/groups/${id}`, {
    method: 'DELETE',
    headers: getAuthHeaders(),
  })
  if (!res.ok) throw new Error('Failed to delete group')
}

export async function fetchGroupModels(groupId: string): Promise<{ items: GroupModel[] }> {
  const res = await fetch(`${API_BASE}/groups/${groupId}/models`, {
    headers: getAuthHeaders(),
  })
  if (!res.ok) throw new Error('Failed to fetch group models')
  return res.json()
}

export async function setGroupModels(groupId: string, modelIds: string[]): Promise<void> {
  const res = await fetch(`${API_BASE}/groups/${groupId}/models`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
      ...getAuthHeaders(),
    },
    body: JSON.stringify({ model_ids: modelIds }),
  })
  if (!res.ok) throw new Error('Failed to set group models')
}
